ui/src/timeMachine/actions/queries.ts

```diff
@@ -195,6 +199,61 @@ export const saveAndExecuteQueries = () => dispatch => {
   dispatch(executeQueries())
 }
 
+export const executeCheckQuery = () => async (dispatch, getState: GetState) => {
+  const state = getState()
+  const {text} = getActiveQuery(state)
+
+  const {
+    orgs: {
+      org: {id: orgID},
+    },
+  } = state
+
+  if (text == '') {
+    dispatch(setQueryResults(RemoteDataState.Done, [], null))
+  }
+
+  try {
+    dispatch(setQueryResults(RemoteDataState.Loading, null, null, null))
+
+    const startTime = Date.now()
+
+    const extern = parse(
+      'import "influxdata/influxdb/monitor"\noption monitor.write = yield'
+    )
+
+    const result = await runQuery(orgID, text, extern).promise
+    const duration = Date.now() - startTime
+
+    if (result.type === 'UNKNOWN_ERROR') {
+      throw new Error(result.message)
+    }
+
+    if (result.type === 'RATE_LIMIT_ERROR') {
+      dispatch(notify(rateLimitReached(result.retryAfter)))
+
+      throw new Error(result.message)
+    }
+
+    if (result.didTruncate) {
+      dispatch(notify(resultTooLarge(result.bytesRead)))
+    }
+
+    checkQueryResult(result.csv)
+
+    const file = result.csv
+
+    dispatch(setQueryResults(RemoteDataState.Done, [file], duration, null))
+  } catch (e) {
+    if (e.name === 'CancellationError') {
+      return
+    }
+
```

Comment:
do we want to alert honeybadger here?

Reply:
nope! query errors are displayed to user and are helpful to understand that there is something legitimately wrong with users query and what it is. There is nothing for us to fix here.